### PR TITLE
Add Node v10-14+ support

### PR DESCRIPTION
**Pull request by milesj**
Was attempting a Node 12 upgrade and this library currently fails to compile.

### CODE REVIEW

File: src/memwatch.cc [hunk 1]
```diff
@@ -216,9 +216,9 @@ NAN_METHOD(memwatch::upon_gc) {
 NAN_METHOD(memwatch::trigger_gc) {
     Nan::HandleScope scope;
     int deadline_in_ms = 500;
-    if (info.Length() >= 1 && info[0]->IsNumber()) {
-    		deadline_in_ms = (int)(info[0]->Int32Value()); 
-    }
+    // if (info.Length() >= 1 && info[0]->IsNumber()) {
```
Comment (milesj) [inline src/memwatch.cc:219]: This currently fails. Haven't been able to figure it out yet (my c++ ain't the best).

```
../src/memwatch.cc:220:24: error: cannot convert 'Maybe<int32_t>' (aka 'Maybe<int>') to 'int' without a conversion operator
                deadline_in_ms = (int)(info[0]->Int32Value(Nan::GetCurrentContext()));
                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
make: *** [Release/obj.target/memwatch/src/memwatch.o] Error 1
```

Comment (esprehn) [reply src/memwatch.cc:219]: Int32Value returns a Maybe<uint32_t>, you need to do `.FromJust()` or do a checked conversion if you trust it. You can also do a FromMaybe(default):
https://v8docs.nodesource.com/node-12.0/d9/d4b/classv8_1_1_maybe.html#ad8fadc9281e13bbb6de0b7a504dcf2a3

Comment (milesj) [reply src/memwatch.cc:219]: Awesome, `FromJust` worked.

File: src/heapdiff.cc [hunk 2]
```diff
@@ -53,7 +53,9 @@ heapdiff::HeapDiff::Initialize ( v8::Handle<v8::Object> target )
 
     Nan::SetPrototypeMethod(t, "end", End);
 
-    target->Set(Nan::New<v8::String>("HeapDiff").ToLocalChecked(), t->GetFunction());
+    Nan::Set(target, Nan::New<v8::String>("HeapDiff").ToLocalChecked(), t->GetFunction(
```
[the remark this answers is not in the document]
Comment (milesj) [reply src/heapdiff.cc:56]: @esprehn Tried that but it fails. I _think_ that `Set` is closer to the original intention.

[the remark this answers is not in the document]
Comment (esprehn) [reply src/heapdiff.cc:56]: Yeah I see now what this is doing. target is actually exports of the module, and this is making HeapDiff the export. There's nicer ways to do all this, but I'm not sure it's worth worrying about right now.

**Comment by OmgImAlexis** [comment]
I would have opened an issue but this is a fork so thank Github.

I tried using this with node v14 and this I'm just hitting a segfault. I don't have much more to go off of.

This is the code I was trying.
```ts
// Once a minute check the memory diff
try {
    let lastHeap = new memwatch.HeapDiff();
    lastHeap.end();
} catch (error: unknown) {
    console.log('failed to get gc diff with "%"', (error as Error).message);
}
```

```ts
[debug] [@unraid]: Loading MyServers API key... [./dist/index.js:21476]
PID 30564 received SIGSEGV for address: 0x400000018
/usr/local/bin/node/unraid-api/node_modules/segfault-handler/build/Release/segfault-handler.node(+0x3276)[0x14e1df447276]
/lib64/libpthread.so.0(+0x13690)[0x14e1e004d690]
unraid-api(_ZN2v88internal14V8HeapExplorer24ExtractContextReferencesEPNS0_9HeapEntryENS0_7ContextE+0xd6)[0x104ba06]
unraid-api(_ZN2v88internal14V8HeapExplorer27IterateAndExtractReferencesEPNS0_21HeapSnapshotGeneratorE+0x385)[0x1050f75]
unraid-api(_ZN2v88internal21HeapSnapshotGenerator16GenerateSnapshotEv+0xd4)[0x1051c14]
unraid-api(_ZN2v88internal12HeapProfiler12TakeSnapshotEPNS_15ActivityControlEPNS_12HeapProfiler18ObjectNameResolverEb+0x6f)[0x1041bef]
/usr/local/bin/node/unraid-api/node_modules/@airbnb/node-memwatch/build/Release/memwatch.node(_ZN8heapdiff8HeapDiff3NewERKN3Nan20FunctionCallbackInfoIN2v85ValueEEE+0x10f)[0x14e1df439e4f]
/usr/local/bin/node/unraid-api/node_modules/@airbnb/node-memwatch/build/Release/memwatch.node(+0x59c8)[0x14e1df4399c8]
unraid-api(_ZN2v88internal25FunctionCallbackArguments4CallENS0_15CallHandlerInfoE+0x156)[0xc10366]
unraid-api[0xc10c31]
unraid-api[0xc10f2a]
unraid-api(_ZN2v88internal21Builtin_HandleApiCallEiPmPNS0_7IsolateE+0x16)[0xc11756]
unraid-api[0x1483a19]
/etc/rc.d/rc.unraid-api: line 83: 30564 Segmentation fault      ENVIRONMENT=$(echo $env) DEBUG=true node $node_base_directory/unraid-api/index.js
```

**Pull request merged**